clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1538,14 +1546,29 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
         acquireAndEnsureOpen();
         try {
             maybeThrowInvalidGroupIdException();
+            maybeThrowIfCommitOffsetsNotOwned(offsets);
             log.debug("Committing offsets: {}", offsets);
             offsets.forEach(this::updateLastSeenEpochIfNewer);
             coordinator.commitOffsetsAsync(new HashMap<>(offsets), callback);
+        } catch (CommitFailedException e) {
+            log.error("Failed to commit offsets asynchronously because they do not belong to dynamically assigned partitions");
```

Comment:
Just wondering, does this mean we did not used to check if all the offsets actually belonged to auto-assigned partitions? Or did we just straight throw the `CommitFailedException`? Is this a behavioral change?

Reply:
I think this is not a behavior change: before this, the broker would check the commit offset request (but only on the generation id, as the coordinator did not keep trace who-owns-what), and if failed it will return the error code which would be translated to CommitFailed on the callback still.

With this we will set the same error on the callback, but much earlier and skip the offset commit request.

Reply:
Do we still need this `catch` since we reverted the logic to verify only assigned partitions can be committed?